packages/framework/src/Command/MigrateImagesCommand.php

```diff
@@ -0,0 +1,236 @@
+     */
+    protected function deleteAllLegacyImageSizesDirectories(array $directories, SymfonyStyle $style): array
+    {
+        foreach ($directories as $key => $directory) {
+            if (basename($directory) === 'original') {
+                continue;
+            }
+
+            $directoryCanBeRemoved = $this->isDeepestDirectory($directories, $directory);
+
+            if (!$directoryCanBeRemoved) {
+                continue;
+            }
+
+            $this->deleteDirectory($directory, $style);
+            unset($directories[$key]);
+        }
+
+        return $directories;
+    }
+
+    /**
+     * @param string[] $directories
+     * @return string[]
+     */
+    private function sortDirectoriesFromDeepestLevel(array $directories): array
+    {
+        usort($directories, static function ($directoryA, $directoryB) {
+            // the more directory separators there are in the path, the deeper the directory is
+            return preg_match_all(static::DIRECTORY_SEPARATOR_REGEX, $directoryB) <=> preg_match_all(static::DIRECTORY_SEPARATOR_REGEX, $directoryA);
+        });
+
+        return $directories;
+    }
+
+    /**
+     * @param string[] $directories
+     * @return string[]
+     */
+    private function getOriginalSizeDirectories(array $directories): array
+    {
+        return array_filter($directories, static function ($directory) {
+            return basename($directory) === 'original';
+        });
+    }
+
+    /**
+     * @return string
+     */
+    protected function getImageEntityNamesRegex(): string
+    {
+        $imageEntityNames = array_map(static fn (ImageEntityConfig $imageEntityConfig) => ($imageEntityConfig->getEntityName()), $this->imageConfig->getAllImageEntityConfigsByClass());
+
+        return '@' . implode('|', $imageEntityNames) . '@';
+    }
+
+    /**
+     * @param string $directory
+     * @return string[]
+     */
+    protected function listDirectoriesRecursive(string $directory): array
+    {
+        $directories = [];
+
+        foreach ($this->filesystem->listContents($directory) as $fileInfo) {
+            if ($fileInfo->isDir()) {
+                $directories[] = $fileInfo->path();
+                $subDirectories = $this->listDirectoriesRecursive($fileInfo->path());
+                $directories = [...$directories, ...$subDirectories];
+            }
+        }
+
+        return $directories;
+    }
+
+    /**
+     * @param string $directory
+     * @param \Symfony\Component\Console\Style\SymfonyStyle $style
+     */
+    private function deleteDirectory(string $directory, SymfonyStyle $style): void
+    {
+        $isDirectoryEmpty = $this->filesystem->listContents($directory)->toArray() === [];
+        $additionalInfo = $isDirectoryEmpty ? 'is empty' : 'with all its content';
+
+        $style->writeln(sprintf('Removing directory %s (%s)', $directory, $additionalInfo));
+
+        if ($isDirectoryEmpty) {
+            $this->filesystem->delete($directory . '/');
```

Comment:
deleting the empty directory on the local filesystem will result in an error.
![Snímek obrazovky 2023-11-29 v 17 15 56](https://github.com/shopsys/shopsys/assets/1177414/82de8907-24f9-4474-908d-dedd30810677)

I understand that the migration script is meant primarily for the S3 storage, but as it's a part of the deploy phing target, I don't think we can ignore that

Reply:
I am not happy about that but here is a workaround for the problem - 4443ed998

Reply:
nice